sources/api/netdog/src/net_config.rs

```diff
@@ -0,0 +1,197 @@
+//! The net_config module contains the strucures needed to deserialize a `net.toml` file.
+//!
+//! These structures are the user-facing options for configuring one or more network interfaces.
+use crate::interface_name::{deserialize_interfaces_map, InterfaceName};
+use indexmap::IndexMap;
+use serde::Deserialize;
+use snafu::{ensure, OptionExt, ResultExt};
+use std::{fs, path::Path};
+
+pub(crate) static NET_CONFIG: &str = "/var/lib/bottlerocket/net.toml";
+
+// TODO: support deserializing different versions of this configuration.
+// Idea: write a deserializer that uses the `version` field and deserializes the rest of the config
+// into an enum with variants for each version, i.e.
+// enum NetConfig {
+//     V1(NetInterfaceV1)
+//     V2(NetInterfaceV2)
+// }
+#[derive(Debug, Deserialize)]
+pub(crate) struct NetConfig {
+    pub(crate) version: u8,
+    // Use an IndexMap to preserve the order of the devices defined in the net.toml.  The TOML
+    // library supports this through a feature making use of IndexMap.  Order is important because
+    // we use the first device in the list as the primary device if the `primary` key isn't set for
+    // any of the devices.
```

Comment:
I thought about that as well.  I think it could start to get a little confusing in the case of interface names like `enp5s0`.  I like the idea of eliminating the dependency, but feel choosing the first in line is pretty clear.